spacy_llm/tests/tasks/test_ner.py

```diff
@@ -323,7 +323,7 @@ def test_ensure_offsets_correspond_to_substrings(
 )
 def test_ner_zero_shot_task(text, response, gold_ents):
     labels = "PER,ORG,LOC"
-    llm_ner = make_ner_task_v2(labels=labels)
+    llm_ner = make_ner_task_v3(labels=labels)
```

Comment:
I'm surprised that only the task factory is changed in these tests, and the test themselves have remained unchanged. Doesn't the `response` look quite different with the v3 task? Where are we testing that the new format is actually parsed correctly on a realistic end-to-end example?

Reply:
> I'm surprised that only the task factory is changed in these tests, and the test themselves have remained unchanged. 

As long as the examples have been adjusted to include span reasons, this should work.

> Where are we testing that the new format is actually parsed correctly on a realistic end-to-end example?

What do you consider realistic? This test includes a few IMO realistic sentences. We could be more thorough, but this is sufficient in my opinion.

Reply:
You're right that in terms of end-to-end example this should be sufficient. I think I was confused also by the fact that we have tests that parse the actual prompt to check the text is formatted as we expect. I don't see any of those tests added for v3?